Occupation evaluation

lizgzil opened the pull request:
---

# Description

Also includes the script of where I created the sample for the skills analysis!


Fixes #73 

# Instructions for Reviewer

In order to test the code in this PR you need to ...

Please pay special attention to ...

# Checklist:

- [ ] I have refactored my code out from `notebooks/`
- [ ] I have checked the code runs
- [ ] I have tested the code
- [ ] I have run `pre-commit` and addressed any issues not automatically fixed
- [ ] I have merged any new changes from `dev`
- [ ] I have documented the code
  - [ ] Major functions have docstrings
  - [ ] Appropriate information has been added to `README`s
- [ ] I have explained this PR above
- [ ] I have requested a code review

india-kerle commented on dap_prinz_green_jobs/pipeline/evaluation/skills_evaluation_sample.py:
```diff
@@ -0,0 +1,96 @@
+        lambda x: x.sample(500, random_state=42)
+    )
+    skills_df_sample.drop(columns=["ENTS", "GREEN_ENTS", "all_esco_map"], inplace=True)
+    skills_df_sample.to_csv(
+        "dap_prinz_green_jobs/pipeline/evaluation/skill_evaluation_sample.csv"
+    )
```
might want to save with the parameter `index=False` so you don't need to deal with an unnamed: 0 column

lizgzil commented on dap_prinz_green_jobs/pipeline/green_measures/occupations/evaluation/README.md:
```diff
@@ -0,0 +1,105 @@
+# 🤔 Evaluation
```
Nice - I'll add it in, also the LM4A comparison (which I did a bit of during a previous iteration but havent repeated)

lizgzil commented on dap_prinz_green_jobs/pipeline/green_measures/occupations/evaluation/socmapper_evaluation_sample.py:
```diff
@@ -0,0 +1,123 @@
+    )
+    common_job_titles = count_job_titles[0:1000].to_dict()
+
+    random_job_titles = random.sample(list(count_job_titles.items()), 200)
+    random_job_titles = {k: v for (k, v) in random_job_titles}
```
oh god - that's bad I didn't, thanks for picking up! I have added one, but I guess it's kind of too late

lizgzil commented on dap_prinz_green_jobs/pipeline/green_measures/occupations/evaluation/socmapper_evaluation_sample.py:
```diff
@@ -0,0 +1,123 @@
+        soc_mapper, common_job_titles, soc_2020_ext_dict, soc_2020_dict
+    )
+
+    common_evaluation_data_df.to_csv(
+        "dap_prinz_green_jobs/pipeline/green_measures/occupations/evaluation/soc_evaluation_sample.csv"
+    )
+    random_evaluation_data_df = get_evaluation_df(
+        soc_mapper, random_job_titles, soc_2020_ext_dict, soc_2020_dict
+    )
+
+    random_evaluation_data_df.to_csv(
+        "dap_prinz_green_jobs/pipeline/green_measures/occupations/evaluation/soc_evaluation_random_sample.csv"
+    )
```
good point

lizgzil commented on dap_prinz_green_jobs/pipeline/green_measures/occupations/evaluation/socmapper_evaluation_sample.py:
```diff
@@ -0,0 +1,123 @@
+"""
```
will do!

lizgzil commented:
Thanks @india-kerle - I've addressed your comments.

I think when we have merged all evaluation stuff we can create a better structure for this folder - e.g. at the moment the occupation evaluation samples are in one place, with the evaluation notebook and README in another.
